test/utils/connectMockContract.ts

```diff
@@ -0,0 +1,170 @@
+// Connect to mock contract without deploying it
+// We should add this to Waffle and remove from here
```

Comment:
+1

This is pretty awesome! Scary to review as part of a large PR though. Nothing jumps out to me on a cursory read-through, but we should take a closer look post-merge to understand what's going on here.

Reply:
It's mostly copied code from Waffle

Reply:
The only thing changed is removing of mock contract deployment

Reply:
Thanks for the background! I guess this is also a test util, so the scope of possible damage isn't as great as modifying Solidity code. LGTM.